sklearn/utils/_plotting.py

```diff
@@ -250,6 +251,125 @@ def _validate_curve_kwargs(
+        )
+        self.line_tooltip_.set_visible(False)
+
+        # Set an attribute on the axes annotation to be able to keep only one visible
+        # at a time when there are multiple display instances that share an axes.
+        setattr(self.line_tooltip_, "_skl_line_tooltip", True)
+
+        self.ax_.figure.canvas.mpl_connect(
+            "motion_notify_event", partial(self._hover, t_vals=t_vals)
+        )
+
+    def _hover(self, event, t_vals):
+        """Callback for the mouse over event.
+
+        Parameters
+        ----------
+        event : matplotlib event
+            the event triggering the callback.
+
+        t_vals : list of ndarrays or None
+            The parameter values along the line.
+        """
+        if event.inaxes != self.ax_:
+            return
+
+        lines = _convert_to_list_leaving_none(self.line_)
+        for i, line in enumerate(lines):
+            contains, indexes = line.contains(event)
+            # stop at the first line on which the event occured
+            if contains:
+                idx = indexes["ind"][0]
+                x_vals, y_vals = line.get_data()
+                t = t_vals[i][idx] if t_vals is not None else t_vals
+                self._update_line_tooltip(x_vals[idx], y_vals[idx], t)
+                self.line_tooltip_.set_visible(True)
+                break
+        else:  # hide the tooltip if the event didn't occur on any line
+            if self.line_tooltip_.get_visible():
+                self.line_tooltip_.set_visible(False)
+
+        # Loop through all line tooltips contained the axes and hide all but one.
+        # This is necessary in addition to the loop over the display lines above to
+        # account for the case when multiple display instances share the same axes.
+        found_visible = False
+        for child in self.ax_.get_children():
+            if hasattr(child, "_skl_line_tooltip") and child.get_visible():
+                if not found_visible:
+                    found_visible = True
+                else:
+                    child.set_visible(False)
+
+        self.figure_.canvas.draw_idle()
+
+    def _update_line_tooltip(self, x, y, t):
+        """Update the text in the line tooltip.
+
+        Parameters
+        ----------
+        x : float
+            the x value to display.
+
+        y : float
+            the y value to display.
+
+        t : float or None
+            the parameter value to display.
+        """
+        text = f"{self._x_label_short}: {x:.2f}, {self._y_label_short}: {y:.2f}"
+        if t is not None:
+            text += f", {self._t_label}: {t:.2f}"
+
+        # Compute an offset for the text depending on the quadrant where the cursor is
+        # to keep the tooltip somewhat inside the axes.
+        xlim, ylim = self.ax_.get_xlim(), self.ax_.get_ylim()
+        x_offset = 20 if x < (xlim[0] + xlim[1]) / 2 else -160
+        y_offset = 20 if y < (ylim[0] + ylim[1]) / 2 else -30
```

Comment:
These hardcoded numbers are not perfect for all cases. We can have a constant offset and let the tooltip be outside the axes sometimes, or I can try to find a way to use the text length to have a matching offset.